tests/suite/scale_test.go

```diff
@@ -728,66 +796,360 @@ var _ = Describe("Scale test", Ordered, Label("nfr", "scale"), func() {
+		Expect(framework.WriteSystemInfoToFile(outFile, clusterInfo, *plusEnabled)).To(Succeed())
+	})
+
+	AfterAll(func() {
+		_, err := fmt.Fprintln(outFile)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(outFile.Close()).To(Succeed())
+
+		// restoring NGF shared among tests in the suite
+		cfg := getDefaultSetupCfg()
+		cfg.nfr = true
+		setup(cfg)
+	})
+
+	BeforeEach(func() {
+		ns = core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "zero-downtime-scale",
+			},
+		}
+
+		metricsCh = make(chan *metricsResults, 2)
+	})
+
+	tests := []struct {
+		name        string
+		valuesFile  string
+		numReplicas int
+	}{
+		{
+			name:        "One NGF Pod runs per node",
+			valuesFile:  "manifests/scale/zero-downtime/values-affinity.yaml",
+			numReplicas: 12, // equals number of nodes
+		},
+		{
+			name:        "Multiple NGF Pods run per node",
+			valuesFile:  "manifests/scale/zero-downtime/values.yaml",
+			numReplicas: 24, // twice the number of nodes
+		},
+	}
+
+	for _, test := range tests {
+		When(test.name, func() {
+			BeforeAll(func() {
+				cfg := getDefaultSetupCfg()
+				cfg.nfr = true
+				setup(cfg, "--values", test.valuesFile)
+
+				deploy, err := resourceManager.GetNGFDeployment(ngfNamespace, releaseName)
+				Expect(err).ToNot(HaveOccurred())
+				ngfDeploymentName = deploy.GetName()
+
+				Expect(resourceManager.Apply([]client.Object{&ns})).To(Succeed())
+				Expect(resourceManager.ApplyFromFiles(files, ns.Name)).To(Succeed())
+				Expect(resourceManager.WaitForAppsToBeReady(ns.Name)).To(Succeed())
+
+				_, err = fmt.Fprintf(outFile, "\n## %s Test Results\n", test.name)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterAll(func() {
+				teardown(releaseName)
+				Expect(resourceManager.DeleteNamespace(ns.Name)).To(Succeed())
+			})
+
+			It("scales up gradually without downtime", func() {
+				_, err := fmt.Fprint(outFile, "\n### Scale Up Gradually\n")
+				Expect(err).ToNot(HaveOccurred())
+
+				testFileNamePrefix := formatTestFileNamePrefix("gradual-scale-up", test.valuesFile)
+
+				sendTraffic(testFileNamePrefix, 5*time.Minute)
+
+				// allow traffic flow to start
+				time.Sleep(2 * time.Second)
+
+				// scale NGF up one at a time
+				for i := 2; i <= test.numReplicas; i++ {
+					Eventually(resourceManager.ScaleDeployment).
+						WithArguments(ngfNamespace, ngfDeploymentName, int32(i)).
+						WithTimeout(timeoutConfig.UpdateTimeout).
+						WithPolling(500 * time.Millisecond).
+						Should(Succeed())
+
+					gatewayFile := fmt.Sprintf("scale/zero-downtime/gateway-%d.yaml", i)
+					Expect(resourceManager.ApplyFromFiles([]string{gatewayFile}, ns.Name)).To(Succeed())
+
+					ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.UpdateTimeout)
+
+					Expect(resourceManager.WaitForPodsToBeReadyWithCount(ctx, ngfNamespace, i)).To(Succeed())
+					Expect(resourceManager.WaitForGatewayObservedGeneration(ctx, ns.Name, "gateway", i)).To(Succeed())
+
+					cancel()
+				}
+
+				wg.Wait()
```

Comment:
This seems disconnected from the test and like it would be very easy to forget in the future. 

Can we move the wait group functionality into the test case so it's more obvious?